compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3466,6 +3466,31 @@ boolean checkLiteralAssignabilityBasedOnType(BLangLiteral baseLiteral, BLangLite
         return false;
     }
 
+    boolean validateFloatLiteral(Location pos, String numericLiteral) {
+        double value = Double.parseDouble(numericLiteral);
+        if (Double.isInfinite(value)) {
+            dlog.error(pos, DiagnosticErrorCode.FLOAT_TOO_LARGE, numericLiteral);
+            return false;
+        }
+        if (value != 0.0) {
+            return true;
+        }
+
+        var exponentIndicator = List.of('e', 'E', 'p', 'P');
+        for (int i = 0; i < numericLiteral.length(); i++) {
```

Comment:
This doesn't seems to be efficient. Let's handle it in a regex as we discussed before. Let's handle it seperately

Reply:
Create an issue(https://github.com/ballerina-platform/ballerina-lang/issues/33795)